Library/Homebrew/test/utils_spec.rb

```diff
@@ -0,0 +1,340 @@
+require "utils"
+require "tempfile"
+require "utils/shell"
+
+RSpec::Matchers.alias_matcher :have_failed, :be_failed
+
+def esc(code)
+  /(\e\[\d+m)*\e\[#{code}m/
+end
+
+module Utils
+  describe self do
+    let(:dir) { Pathname.new(Dir.mktmpdir) }
+
+    specify "#ofail" do
+      begin
+        shutup do
+          ofail "foo"
+        end
+
+        expect(Homebrew).to have_failed
+      ensure
+        Homebrew.failed = false
+      end
+    end
+
+    specify "#odie" do
+      expect(self).to receive(:exit).and_return(1)
+      shutup do
+        odie "foo"
+      end
+    end
+
+    describe "#pretty_installed" do
+      context "with HOMEBREW_NO_EMOJI unset and when $stdout is a TTY" do
```

Comment:
"when $stdout is a tty" could be its own context and the stub would only have to be specified once.